src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIMarsManager.cs

```diff
@@ -12,14 +11,14 @@ namespace System.Data.SqlClient.SNI
     /// </summary>
     internal class SNIMarsManager
     {
-        public static readonly SNIMarsManager Singleton = new SNIMarsManager();
-        private ConcurrentDictionary<SNIHandle, SNIMarsConnection> _connections = new ConcurrentDictionary<SNIHandle, SNIMarsConnection>();
+        private readonly ConcurrentDictionary<SNIHandle, SNIMarsConnection> _connections;
 
         /// <summary>
         /// Constructor
         /// </summary>
         public SNIMarsManager()
```

Comment:
So is it ok for us to be able to have multiple instance of `SNIMarsManager`?
This code change removes Singleton design.